rpc/src/v1/impls/ethcore.rs

```diff
@@ -115,6 +118,47 @@ fn vm_trace_to_object(t: &VMTrace) -> Value {
 	Value::Object(ret)
 }
 
+fn diff_to_object<T>(d: &Diff<T>) -> Value where T: serde::Serialize + Eq {
+	let mut ret = BTreeMap::new();
+	match *d {
+		Diff::Same => {
+			ret.insert("diff".to_owned(), Value::String("=".to_owned()));
+		}
+		Diff::Born(ref x) => {
+			ret.insert("diff".to_owned(), Value::String("+".to_owned()));
+			ret.insert("+".to_owned(), to_value(x).unwrap());
+		}
+		Diff::Died(ref x) => {
+			ret.insert("diff".to_owned(), Value::String("-".to_owned()));
+			ret.insert("-".to_owned(), to_value(x).unwrap());
+		}
+		Diff::Changed(ref from, ref to) => {
+			ret.insert("diff".to_owned(), Value::String("*".to_owned()));
+			ret.insert("-".to_owned(), to_value(from).unwrap());
+			ret.insert("+".to_owned(), to_value(to).unwrap());
+		}
+	};
+	Value::Object(ret)
+}
+
+fn state_diff_to_object(t: &StateDiff) -> Value {
+	Value::Object(t.iter().map(|(address, account)| {
+		(address.hex(), Value::Object(map![
+			"existance".to_owned() => Value::String(match account.existance() {
+				Existance::Born => "+",
+				Existance::Alive => ".",
+				Existance::Died => "-",
+			}.to_owned()),
+			"balance".to_owned() => diff_to_object(&account.balance),
+			"nonce".to_owned() => diff_to_object(&account.nonce),
+			"code".to_owned() => diff_to_object(&account.code),
+			"storage".to_owned() => Value::Object(account.storage.iter().map(|(key, val)| {
+				(key.hex(), diff_to_object(&val))
+			}).collect::<BTreeMap<_, _>>())
+		]))
+	}).collect::<BTreeMap<_, _>>())
```

Comment:
Can you move it to a separate data structure utilizing serde's `Serialize` trait and add some tests? It would be more consistent with other data structures in rpc.